File: ui.cyberminer/WebContent/LineStorage.js
var LineStorage = function () {
	this.lines = [];
};

LineStorage.prototype.setLines = function(lines){
	this.lines = lines;
};

LineStorage.prototype.getLine = function(i){
	return this.lines[i];
};

LineStorage.prototype.getLines = function(){
	return this.lines;
};

File: web.cyberminer/src/main/java/edu/utd/sbrp/web/cyberminer/module/LineStorage.java
package edu.utd.sbrp.web.cyberminer.module;

import java.util.ArrayList;
import java.util.List;

public class LineStorage implements IndexModule {
	
	List<String> lines = new ArrayList<String>();
	
	public LineStorage() {
	}
	
	public void process() {
		// not used
	}

	public void setLine(int l, String line) {
		// fill it with null
		int numToFill = (l - lines.size());
		
		for(int i = 0; i < numToFill; ++i) {
			lines.add(null); // init lines up to l-th line 
		}
		
		lines.add(line);
	}


	public String getLine(int l) {
		String line = null;
		try {
			line = lines.get(l);
		} catch (IndexOutOfBoundsException e) {
			// do nothing...
		}
		return line;
	}

}
